verifier/verifier.go

```diff
@@ -1019,7 +1087,27 @@ func verifyTimestamp(ctx context.Context, policyName string, trustStores []strin
 
 	// 5. Perform the timestamping certificate chain revocation check
 	logger.Debug("Checking timestamping certificate chain revocation...")
-	certResults, err := r.Validate(tsaCertChain, time.Time{})
+
+	var certResults []*revocationresult.CertRevocationResult
+	switch revocationMode {
+	case trustpolicy.RevocationModeAuto:
+		logger.Debug("Revocation mode is set to auto")
+		certResults, err = r.Validate(tsaCertChain, time.Time{})
+	case trustpolicy.RevocationModeOCSP:
+		logger.Debug("Revocation mode is set to OCSP")
+		if r, ok := r.(revocation.OCSPRevocation); ok {
+			certResults, err = r.ValidateOCSPOnly(tsaCertChain, time.Time{})
+		} else {
+			return errors.New("revocation client does not support OCSP")
+		}
+	case trustpolicy.RevocationModeCRL:
+		logger.Debug("Revocation mode is set to CRL")
+		if r, ok := r.(revocation.CRLRevocation); ok {
+			certResults, err = r.ValidateCRLOnly(tsaCertChain, time.Time{})
+		} else {
+			return errors.New("revocation client does not support CRL")
+		}
+	}
```

Comment:
Do we need a default case here?

Reply:
Yes, we need default. This PR is for PoC only. `RevocationMode` will no be supported in v1.3.0